platform/ui/src/utils/ModalProvider.js

```diff
@@ -1,70 +1,53 @@
 import React, { useState, createContext, useContext } from 'react';
-import Modal from 'react-modal';
+import Modal from 'react-bootstrap-modal';
 
 const ModalContext = createContext(null);
+const { Provider, Consumer } = ModalContext;
 
 export const useModalContext = () => useContext(ModalContext);
 
-/* Bind modal to app root (http://reactcommunity.org/react-modal/accessibility/). */
-Modal.setAppElement('#root');
-
-const computedStyle = getComputedStyle(document.body);
-const customStyles = {
-  content: {
-    top: '50%',
-    left: '50%',
-    right: 'auto',
-    bottom: 'auto',
-    marginRight: '-50%',
-    transform: 'translate(-50%, -50%)',
-    backgroundColor: computedStyle.getPropertyValue('--ui-gray-darker'),
-    color: computedStyle.getPropertyValue('--text-secondary-color'),
-    borderColor: computedStyle.getPropertyValue('--ui-border-color'),
-    border: 0,
-    borderRadius: '6px',
-  },
-  overlay: {
-    background: 'rgba(0, 0, 0, 0.5)',
-  },
-};
-
 const ModalProvider = ({ children }) => {
   const DEFAULT_OPTIONS = {
     component: null,
     props: {
-      shouldCloseOnOverlayClick: false,
-      shouldCloseOnEsc: false,
+      backdrop: false,
+      keyboard: false,
+      show: true,
+      large: true,
+      closeButton: true,
+      title: null,
     },
   };
 
   const [options, setOptions] = useState(DEFAULT_OPTIONS);
 
-  const show = (component, props = {}) => {
-    setOptions({ ...options, component, props: { ...options.props, props } });
-  };
+  const show = (component, props = {}) =>
+    setOptions({
+      ...options,
+      component,
+      props: { ...options.props, ...props },
+    });
 
-  const hide = () => {
-    setOptions({ ...options, component: null, props: {} });
-  };
+  const hide = () => setOptions(DEFAULT_OPTIONS);
 
   return (
-    <ModalContext.Provider value={{ ...options, show, hide }}>
-      <ModalContext.Consumer>
+    <Provider value={{ ...options, show, hide }}>
+      <Consumer>
         {({ component: Component, props, hide }) =>
           Component ? (
-            <Modal
-              style={customStyles}
-              isOpen={true}
-              {...props}
-              onRequestClose={hide}
-            >
-              <Component {...props} onRequestClose={hide} />
+            <Modal {...props} onHide={hide}>
+              {props.title && (
+                <Modal.Header closeButton={props.closeButton}>
+                  <Modal.Title>{props.title}</Modal.Title>
+                </Modal.Header>
+              )}
+              <Component {...props} hide={hide} />
             </Modal>
           ) : null
         }
-      </ModalContext.Consumer>
+      </Consumer>
       {children}
```

Comment:
I don't think we want `{children}` here. Let's be more explicit.

You can provide a `title` or `HeaderComponent`
`body` or `BodyComponent`
`footer` (more complex) or `FooterComponent`

Children's a bit ambiguous as we have named (multiple) insertion points